src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonWebToken.PayloadClaimSet.cs

```diff
@@ -89,7 +89,7 @@ internal JsonClaimSet CreatePayloadClaimSet(byte[] bytes, int length)
                     }
                     else if (reader.ValueTextEquals(JwtPayloadUtf8Bytes.Sub))
                     {
-                        _sub = JsonSerializerPrimitives.ReadString(ref reader, JwtRegisteredClaimNames.Sub, ClassName, true);
+                        _sub = JsonSerializerPrimitives.ReadAsString(ref reader, JwtRegisteredClaimNames.Sub, ClassName, true);
```

Comment:
could it be ReadNumber(ref reader).ToString() here instead of create a new parse method?

Reply:
I don't think so, ReadNumber only checks reader it's a valid number and returns it back, Although, I see why ToString() might seem as a way to do it, there's additional logic that deals with Null as well as, additional exceptions we throw when dealing with Array, Object and Bool types.